src/components/Combobox.tsx

```diff
@@ -78,7 +79,7 @@ function Combobox<T>({
     }
     return optionsProp as Option<T>[];
   }, [optionsProp, grouped]);
-  const selected = useMemo<Option<T> | Option<T>[]>(() => multi ? (value || []).map((v: T) => options.find(option => option.value === v)) : options.find(option => option.value === value), [value, options, multi]);
+  const selected = useMemo<Option<T> | Option<T>[]>(() => multi ? (value || []).map((v: T) => options.find(option => equal(option.value, v))) : options.find(option => equal(option.value, value)), [value, options, multi]);
```

Comment:
Do we need this? Seems like the order of the JSON.stringify() should be equivalent each time.

Reply:
If we do use JSON.stringify() here, we could reuse as well for the dropdown.

Reply:
Also, using equal each time is something we should be able to cache when we first pass the options in.  We shouldn't do this each time we have a selection.

Reply:
I feel better actually doing a deep comparison of the object because we don't know what might happen in the external `onChange` handlers. It is possible for the order of the object keys in the `value` prop to be different than the option value.

Reply:
> Also, using equal each time is something we should be able to cache when we first pass the options in. We shouldn't do this each time we have a selection.

`useMemo` should actually take care of this. It's only recomputed when dependencies change.